src/decompiler-ls.h

```diff
@@ -84,18 +102,19 @@ struct LoadStoreTracking {
   };
 
   void Track(const Node& n) {
-    for (auto& c : n.children) Track(c);
+    for (auto& c : n.children)
+      Track(c);
```

Comment:
Does chromium style not require braces around loop blocks?

Reply:
Requiring braces is the area of clang-tidy, not clang-format. We [do that in .clang-tidy](https://github.com/WebAssembly/binaryen/blob/6a2d7f989065820476268a2382db2eccf72aadd7/.clang-tidy#L1) in Binaryen. clang-format's job is to format whatever code given, as long as it compiles.

We can consider adding .clang-tidy to wabt, which should be a follow-up task. In this PR I think we should decide whether we should allow one-line short `if` or `for`/`while`. The current codebase seems to be using one-line `if`s a lot, so I'll try tweaking .clang-format to allow that. Let me know if you don't prefer that way.

Reply:
IIUC chromium/google style (which we are basing wabt on) is OK with single line block (where the keyword and the body are on a single line), but if you have a body on a line by itself it requires braces (unlike LLVM)

See https://google.github.io/styleguide/cppguide.html#Conditionals

Basically `if (x) foo();` is ok, but not

```
if (x) 
  foo();
```

Reply:
Given that I'm a little surprised that clang-format did this here...

Reply:
Actually I think I read that wrong... for historical reasons single line block are ok... ignore me.

Reply:
Sorry, I'm in the middle of fixing things, so I haven't actually committed the newly formatted code with new .clang-format. Sorry for the confusion. I added `AllowShortLoopsOnASingleLine: true`, so this will be back to `if (x) foo();`. I will commit all those changes I guess in an hour or something.